Fix test_price_history.py::test_price_queries in the CI

## Problem Definition

A new test I wrote seems to fail only on the CI for some reason.  Skipped it for now and made this issue so that it does not slow down the rest of the development.

Failing test: https://github.com/rotki/rotki/pull/2036/checks?check_run_id=1665414234

```
2021-01-07T22:47:15.8818595Z =================================== FAILURES ===================================
2021-01-07T22:47:15.8819407Z __________________________ test_price_queries[False] ___________________________
2021-01-07T22:47:15.8819971Z 
2021-01-07T22:47:15.8821239Z price_historian = <rotkehlchen.history.price.PriceHistorian object at 0x7f0f1057cc10>
2021-01-07T22:47:15.8823526Z data_dir = PosixPath('/home/runner/.cache/.rotkehlchen-test-dir')
2021-01-07T22:47:15.8826683Z database = <rotkehlchen.db.dbhandler.DBHandler object at 0x7f0f03f97fd0>
2021-01-07T22:47:15.8827532Z 
2021-01-07T22:47:15.8828832Z     @pytest.mark.parametrize('should_mock_price_queries', [False])
2021-01-07T22:47:15.8829861Z     def test_price_queries(price_historian, data_dir, database):
2021-01-07T22:47:15.8830810Z         """Test some historical price queries. Make sure that we test some
2021-01-07T22:47:15.8831888Z         assets not in cryptocompare but in coigecko so the backup mechanism triggers and works"""
2021-01-07T22:47:15.8832649Z     
2021-01-07T22:47:15.8833196Z         # These should hit cryptocompare
2021-01-07T22:47:15.8834378Z         assert price_historian.query_historical_price(A_BTC, A_EUR, 1479200704) == FVal('663.66')
2021-01-07T22:47:15.8835723Z         assert price_historian.query_historical_price(A_XMR, A_BTC, 1579200704) == FVal('0.007526')
2021-01-07T22:47:15.8836742Z         # this should hit the cryptocompare cache we are creating here
2021-01-07T22:47:15.8837548Z         contents = """{"start_time": 0, "end_time": 1439390800,
2021-01-07T22:47:15.8838249Z         "data": [{"time": 1438387200, "close": 10, "high": 10, "low": 10, "open": 10,
2021-01-07T22:47:15.8839022Z         "volumefrom": 10, "volumeto": 10}, {"time": 1438390800, "close": 20, "high": 20,
2021-01-07T22:47:15.8839784Z         "low": 20, "open": 20, "volumefrom": 20, "volumeto": 20}]}"""
2021-01-07T22:47:15.8840583Z         price_history_dir = get_or_make_price_history_dir(data_dir)
2021-01-07T22:47:15.8841767Z         with open(price_history_dir / f'{PRICE_HISTORY_FILE_PREFIX}DASH_USD.json', 'w') as f:
2021-01-07T22:47:15.8842548Z             f.write(contents)
2021-01-07T22:47:15.8843569Z         price_historian.cryptocompare = Cryptocompare(data_directory=data_dir, database=database)
2021-01-07T22:47:15.8845075Z >       assert price_historian.query_historical_price(A_DASH, A_USD, 1438387700) == FVal('10')
2021-01-07T22:47:15.8845975Z E       AssertionError: assert FVal(3.629) == FVal(10)
2021-01-07T22:47:15.8848774Z E        +  where FVal(3.629) = <function PriceHistorian.query_historical_price at 0x7f0f1668cb90>(<Asset identifier:DASH name:Dash symbol:DASH>, <Asset identifier:USD name:United States Dollar symbol:USD>, 1438387700)
2021-01-07T22:47:15.8852548Z E        +    where <function PriceHistorian.query_historical_price at 0x7f0f1668cb90> = <rotkehlchen.history.price.PriceHistorian object at 0x7f0f1057cc10>.query_historical_price
2021-01-07T22:47:15.8855929Z E        +  and   FVal(10) = FVal('10')
2021-01-07T22:47:15.8856681Z 
2021-01-07T22:47:15.8857995Z rotkehlchen/tests/integration/test_price_history.py:28: AssertionError
2021-01-07T22:47:15.8859839Z travis_fold:start:py-8091.stderr-setup.0
```

## Task

Fix it

--- a/rotkehlchen/tests/integration/test_price_history.py
+++ b/rotkehlchen/tests/integration/test_price_history.py
@@ -6,13 +6,9 @@ from rotkehlchen.fval import FVal
 from rotkehlchen.tests.utils.constants import A_DASH, A_XMR
 from rotkehlchen.utils.misc import get_or_make_price_history_dir
 from rotkehlchen.externalapis.cryptocompare import PRICE_HISTORY_FILE_PREFIX, Cryptocompare
-import os
 
 
-@pytest.mark.skipif(
-    'CI' in os.environ,
-    reason='Need to figure out why this fails only in the CI',
-)
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
 @pytest.mark.parametrize('should_mock_price_queries', [False])
 def test_price_queries(price_historian, data_dir, database):
     """Test some historical price queries. Make sure that we test some
@@ -29,7 +25,10 @@ def test_price_queries(price_historian, data_dir, database):
     price_history_dir = get_or_make_price_history_dir(data_dir)
     with open(price_history_dir / f'{PRICE_HISTORY_FILE_PREFIX}DASH_USD.json', 'w') as f:
         f.write(contents)
-    price_historian.cryptocompare = Cryptocompare(data_directory=data_dir, database=database)
+    price_historian._PriceHistorian__instance._cryptocompare = Cryptocompare(
+        data_directory=data_dir,
+        database=database,
+    )
     assert price_historian.query_historical_price(A_DASH, A_USD, 1438387700) == FVal('10')
     # this should hit coingecko, since cornichon is not in cryptocompare
     cornichon = Asset('CORN-2')

